tests/scripts/all.sh

```diff
@@ -4286,6 +4286,42 @@ component_check_test_helpers () {
     python3 -m unittest tests/scripts/translate_ciphers.py 2>&1
 }
 
+component_test_aes_donot_use_hardware () {
+    msg "Test: AES builtin only"
+    scripts/config.py unset MBEDTLS_AESNI_C
+    scripts/config.py unset MBEDTLS_PADLOCK_C
+    scripts/config.py unset MBEDTLS_AESCE_C
+    scripts/config.py unset MBEDTLS_AES_USE_HARDWARE_ONLY
+    msg "build: make, AES built-in only" # ~10s
+    make
+
+    msg "selftest: AES built-in only" # ~10s
+    programs/test/selftest
```

Comment:
Other `all.sh` components actually check the output from `selftest` for particular strings - should we do that here too?

Reply:
Agreed, please add a component similar to `component_test_aesni` to test the different build options for AESCE.

Reply:
I am re-design the components now

Reply:
Updated. Please check.  

AESCE and VIA Padlock are build only components.  AESCE is build with ArmClang.